mita/async.py

```diff
@@ -61,8 +61,17 @@ def check_idling():
             uuid = n['name'].split('__')[-1]
             if node_info.get('idle'):
                 logging.info("found an idle node: %s" % n['name'])
-                node_endpoint = get_mita_api('nodes', uuid, 'idle')
-                requests.post(node_endpoint)
+                # check to see if this node is still in the db
+                node = models.Node.query.filter_by(identifier=uuid).first()
+                if node:
+                    # node is still in the db, mark it as idle
+                    node_endpoint = get_mita_api('nodes', uuid, 'idle')
+                    requests.post(node_endpoint)
+                else:
```

Comment:
Yes, that is certainly a risk, but I think it's a low one. I guess I'm not sure of a good way around that though, do you have an idea?

We could also not do this and wait to see if we observe the issue again. I'm still not even 100% sure how we get into this state.